op-supervisor/supervisor/backend/status/status.go

```diff
@@ -0,0 +1,59 @@
+package status
+
+import (
+	"sync"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/event"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/superevents"
+)
+
+type StatusTracker struct {
+	statuses map[eth.ChainID]*eth.SupervisorChainStatus
+	mu       sync.RWMutex
+}
+
+func NewStatusTracker() *StatusTracker {
+	return &StatusTracker{
+		statuses: make(map[eth.ChainID]*eth.SupervisorChainStatus),
+	}
+}
+
+func (su *StatusTracker) OnEvent(ev event.Event) bool {
+	su.mu.Lock()
+	defer su.mu.Unlock()
+
+	loadStatus := func(chainID eth.ChainID) *eth.SupervisorChainStatus {
+		v := su.statuses[chainID]
+		if v == nil {
+			v = &eth.SupervisorChainStatus{}
+		}
+		return v
+	}
+	switch x := ev.(type) {
+	case superevents.LocalDerivedEvent:
+		v := loadStatus(x.ChainID)
+		v.LocalDerived = x.Derived.Derived
+		v.LocalDerivedFrom = x.Derived.DerivedFrom
+		su.statuses[x.ChainID] = v
+	case superevents.LocalUnsafeReceivedEvent:
+		v := loadStatus(x.ChainID)
+		v.LocalUnsafe = x.NewLocalUnsafe
+		su.statuses[x.ChainID] = v
```

Comment:
The "Received" event is fired when it is first seen, not when it has completed indexing the unsafe block.
It may be better to wait for the indexing to complete, before updating it as local-unsafe here.

Reply:
Good point. It would be better to reduce confusion. Will do.